### PR TITLE
main: remove log messages when exceptions are thrown

### LINKED ISSUE #5453: main: print error messages to std::cerr from exception text
Currently there are a few places in the code with the pattern:

```
  std::cerr << message;
  throw new EnvoyException(message);
```

These should all be changed to eliminate the stream-out command from the throw site, and instead stream out the exception text from main_common.cc when it's caught.

### PR DESCRIPTION
**Pull request by rishabhkumar296**
Signed-off-by: rishabhkumar296 <kris.kr296@gmail.com>
Fixes #5453 
*Description*: removes std::cerr log messages when exceptions are thrown
*Risk Level*: Low
*Testing*: Unit test
*Docs Changes*: N/A
*Release Notes*: N/A

### CODE REVIEW

File: test/integration/run_envoy_test.sh [hunk 1]
```diff
@@ -22,3 +22,31 @@ expect_fail_with_error "PARSE ERROR: Argument: --bogus-flag" --bogus-flag
 start_test Launching envoy without --config-path or --config-yaml fails.
 expect_fail_with_error \
   "At least one of --config-path and --config-yaml should be non-empty"
+
+start_test Launching envoy with a bogus command line flag.
```
[the remark this answers is not in the document]
Comment (rishabhkumar296) [reply test/integration/run_envoy_test.sh]: Fixed.

File: test/integration/run_envoy_test.sh [hunk 2]
```diff
@@ -22,3 +22,31 @@ expect_fail_with_error "PARSE ERROR: Argument: --bogus-flag" --bogus-flag
...
+expect_fail_with_error "PARSE ERROR: Argument: --bogus-flag" --bogus-flag
+
+start_test Launching envoy without --config-path or --config-yaml fails.
+expect_fail_with_error \
+  "At least one of --config-path and --config-yaml should be non-empty"
```
[the remark this answers is not in the document]
Comment (rishabhkumar296) [reply test/integration/run_envoy_test.sh]: Fixed.

**Comment by rishabhkumar296** [comment]
@jmarantz yes. I tried it with config files as well but it was not required to reproduce the errors.

**Comment by jmarantz** [comment]
Looks good thanks!

**Comment by jmarantz** [comment]
Yup -- the new tests fail in this PR without the std::cerr logging in main.cc.

**Pull request merged**